rfcs/0019-s3 source.md

```diff
@@ -0,0 +1,59 @@
+---
+feature: S3 Source Design
+authors:
+- "Bohan Zhang"
+  start_date: "2022/11/09"
+---
+
+# S3 Source Design
+
+## Motivation
+
+Support load data from Amazon S3 and other Simple Storage Services.
+
+- The previous design is available [here](https://www.notion.so/risingwave-labs/Deprecated-RFC-S3-Source-Design-632aa78264b347f9a4a37fd89a91d785).
+
+## Design
+
+More and more scenarios require us to support data synchronization from S3, which challenges our existing abstractions.
+The main problem in the previous design is that a source could be read many times by multiple materialized views but the SQS message can only be consumed once, which leads to data inconsistency.
+
+In the source abstraction, one source can only have one enumerator to discover the upstream changes. However, the executors in different fragments have different consuming process and different state table.
+It is a trivial idea to discover the changes by every executor and the enumerator only confines the concurrency.
+
+The source executor will start a thread when starting a S3 connector and list all objects in the given bucket.
```

Comment:
Does this mean that every executor will scan whole bucket periodically?

Reply:
yes, we let the executor maintain its own state, just as flink does.